hudi-common/src/main/java/org/apache/hudi/common/table/read/buffer/FileGroupRecordBuffer.java

```diff
@@ -76,7 +72,7 @@ abstract class FileGroupRecordBuffer<T> implements HoodieFileGroupRecordBuffer<T
   protected final RecordMergeMode recordMergeMode;
   protected final PartialUpdateMode partialUpdateMode;
   protected final Option<HoodieRecordMerger> recordMerger;
-  protected final Option<String> payloadClass;
+  protected final Option<Pair<String, String>> payloadClasses;
```

Comment:
can we add java docs as to why we need a Pair of payload classes instead of just 1 entry

Reply:
You can merge an expression payload with a plain avro record essentially based on my understanding of the code but it will need to be validated end to end in the next patch.